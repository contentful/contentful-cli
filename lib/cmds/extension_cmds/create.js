const { handleAsyncError: handle } = require('../../utils/async')
const { createManagementClient } = require('../../utils/contentful-clients')
const { successEmoji } = require('../../utils/emojis')
const { getHeadersFromOption } = require('../../utils/headers')
const { success } = require('../../utils/log')

const { assertExtensionValuesProvided } = require('./utils/assertions')
const { logExtension } = require('./utils/log-as-table')
const prepareData = require('./utils/prepare-data')
const readSrcDocFile = require('./utils/read-srcdoc-file')

module.exports.command = 'create'

module.exports.desc = 'Create an extension'

module.exports.builder = yargs => {
  return yargs
    .option('id', { type: 'string', describe: 'Extension id' })
    .option('name', { type: 'string', describe: 'Extension name' })
    .option('management-token', {
      alias: 'mt',
      type: 'string',
      describe: 'Contentful management API token'
    })
    .option('space-id', { type: 'string', describe: 'Space id' })
    .option('environment-id', { type: 'string', describe: 'Environment id' })
    .option('field-types', { type: 'array', describe: 'Field types' })
    .option('descriptor', {
      type: 'string',
      describe: 'Path to an extension descriptor file'
    })
    .option('src', { type: 'string', describe: 'URL to extension bundle' })
    .option('srcdoc', { type: 'string', describe: 'Path to extension bundle' })
    .option('sidebar', {
      type: 'boolean',
      // We set the default to undefined so the descriptor file value will be
      // used instead of arg value unless explicitly passed in
      default: undefined,
      describe: 'Render the extension in the sidebar'
    })
    .option('installation-parameters', {
      type: 'string',
      describe: 'JSON string of installation parameter key-value pairs'
    })
    .option('header', {
      alias: 'H',
      type: 'string',
      describe: 'Pass an additional HTTP Header'
    })
    .epilog('Copyright 2019 Contentful')
}

async function createExtension(environment, data) {
  const extensionData = { extension: data.extension }
  if (data.parameters) {
    extensionData.parameters = data.parameters
  }
  const createPromise = data.id
    ? environment.createUiExtensionWithId(data.id, extensionData)
    : environment.createUiExtension(extensionData)

  const extension = await createPromise

  return extension
}

module.exports.createExtension = createExtension

async function createExtensionHandler(argv) {
  const data = await prepareData(argv)

  await assertExtensionValuesProvided(data, 'create')

  if (data.extension.srcdoc) {
    await readSrcDocFile(data.extension)
  }

  const { managementToken, activeSpaceId, activeEnvironmentId } = argv.context

  const client = await createManagementClient({
    accessToken: managementToken,
    feature: 'extension-create',
    headers: getHeadersFromOption(argv.header)
  })

  const space = await client.getSpace(activeSpaceId)
  const environment = await space.getEnvironment(activeEnvironmentId)

  const extension = await createExtension(environment, data)

  success(`${successEmoji} Successfully created extension:\n`)

  logExtension(extension, activeSpaceId, activeEnvironmentId)
}

module.exports.createExtensionHandler = createExtensionHandler

module.exports.handler = handle(createExtensionHandler)
