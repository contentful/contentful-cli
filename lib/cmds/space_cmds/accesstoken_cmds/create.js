const { handleAsyncError: handle } = require('../../../utils/async')
const { createManagementClient } = require('../../../utils/contentful-clients')
const { successEmoji } = require('../../../utils/emojis')
const { success } = require('../../../utils/log')
const paginate = require('../../../utils/pagination')
const { highlightStyle } = require('../../../utils/styles')

module.exports.command = 'create'

module.exports.desc = 'Create a delivery access token'

module.exports.builder = yargs => {
  return yargs
    .usage(
      "Usage: contentful accesstoken create --name 'Your token name' --description 'Your token description'"
    )
    .option('name', {
      alias: 'n',
      describe: 'Name of the Token to create',
      demandOption: true
    })
    .option('description', {
      alias: 'desc',
      describe:
        'Description giving more detailed information about the usage of the Token'
    })
    .option('management-token', {
      alias: 'mt',
      describe: 'Contentful management API token',
      type: 'string'
    })
    .option('space-id', { alias: 's', type: 'string', describe: 'Space id' })
    .option('silent', {
      describe: 'Suppress command output',
      default: false
    })
    .epilog('Copyright 2019 Contentful')
}

async function accessTokenCreate(argv) {
  const {
    context,
    name,
    description,
    silent,
    feature = 'space-access_token-create'
  } = argv
  const { managementToken, activeSpaceId } = context

  const client = await createManagementClient({
    accessToken: managementToken,
    feature
  })

  const space = await client.getSpace(activeSpaceId)
  const accessTokens = await paginate({ client: space, method: 'getApiKeys' })

  let accessToken = accessTokens.items.find(key => key.name === name)

  if (accessToken) {
    if (!silent) {
      success(
        `${successEmoji} Successfully returned already existing access token ${highlightStyle(
          accessToken.name
        )} (${highlightStyle(accessToken.accessToken)})`
      )
    }
    return accessToken
  }

  accessToken = await space.createApiKey({
    name,
    description
  })

  if (!silent) {
    success(
      `${successEmoji} Successfully created access token ${highlightStyle(
        accessToken.name
      )} (${highlightStyle(accessToken.accessToken)})`
    )
  }

  return accessToken
}

module.exports.accessTokenCreate = accessTokenCreate

module.exports.handler = handle(accessTokenCreate)
