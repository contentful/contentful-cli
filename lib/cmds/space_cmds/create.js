const logging = require('../../utils/log')
const { handleAsyncError: handle } = require('../../utils/async')
const { createManagementClient } = require('../../utils/contentful-clients')
const { spaceUse } = require('./use')

const { EventSystem } = require('../../core/events')
const { CREATE_SPACE_HANDLER } = require('../../core/events/scopes')

const { warningStyle } = require('../../utils/styles')
const { confirmation } = require('../../utils/actions')
const IntentSystem = require('../../core/event-handlers/intents')
const LoggingSystem = require('../../core/event-handlers/logging')

const createSpaceIntents = require('../../core/event-handlers/intents/create-space-handler')
const createSpaceLogging = require('../../core/event-handlers/logging/create-space-handler')

const { AbortedError } = require('../../guide/helpers')
const { getHeadersFromOption } = require('../../utils/headers')

module.exports.command = 'create'

module.exports.desc = 'Create a space'

module.exports.builder = yargs => {
  return yargs
    .usage("Usage: contentful space create --name 'Your Space Name'")
    .option('name', {
      alias: 'n',
      describe: 'Name of the space to create',
      demandOption: true
    })
    .option('management-token', {
      alias: 'mt',
      describe: 'Contentful management API token',
      type: 'string'
    })
    .option('organization-id', {
      alias: 'org',
      describe: 'Organization owning the new space'
    })
    .option('yes', {
      alias: 'y',
      describe:
        'Confirm space creation without prompt, be aware this may result in extra monthly charges depend on your subscription'
    })
    .option('default-locale', {
      alias: 'l',
      describe: 'The default locale of the new space',
      type: 'string'
    })
    .option('use', {
      alias: 'u',
      describe:
        'Use the created space as default space when the --space-id is skipped.',
      type: 'boolean'
    })
    .option('header', {
      alias: 'H',
      type: 'string',
      describe: 'Pass an additional HTTP Header'
    })
    .epilog('Copyright 2019 Contentful')
}

async function spaceCreate(argv) {
  const {
    context,
    name,
    defaultLocale,
    yes,
    use,
    feature = 'space-create'
  } = argv

  const { managementToken } = context
  let { organizationId } = argv
  const client = await createManagementClient({
    feature,
    accessToken: managementToken,
    headers: getHeadersFromOption(argv.header)
  })

  logging.log(
    warningStyle(`Please be aware that adding new spaces to your subscription,
beyond the free ‘Micro’ space included in each subscription,
will result in extra monthly charges.
More information on available space types and their prices can be found on
the Pricing page: https://www.contentful.com/pricing/?faq_category=payments&faq=what-type-of-spaces-can-i-have#what-type-of-spaces-can-i-have`)
  )

  let confirm = false
  if (!yes) {
    confirm = await confirmation(`Do you want to confirm the space creation?`)
  } else {
    confirm = true
  }

  logging.log()

  if (!confirm) {
    logging.log(warningStyle(`Space creation aborted.`))
    throw new AbortedError()
  }

  const intentSystem = new IntentSystem()
  intentSystem.addHandler(
    createSpaceIntents({
      skipConfirm: yes
    })
  )

  const loggingSystem = new LoggingSystem(logging)
  loggingSystem.addHandler(createSpaceLogging)

  const eventSystem = new EventSystem()
  eventSystem.attachSubsystem(intentSystem)
  eventSystem.attachSubsystem(loggingSystem)

  const dispatcher = eventSystem.dispatcher(CREATE_SPACE_HANDLER)

  if (!organizationId) {
    const result = await client.getOrganizations()

    if (result.items.length > 1) {
      dispatcher.dispatch('MULTIPLE_ORG_MEMBERSHIP')

      const organizationChoices = result.items
        .sort((a, b) => a.name.localeCompare(b.name))
        .map(
          organization => ({
            name: `${organization.name} (${organization.sys.id})`,
            value: organization.sys.id
          }),
          {}
        )

      organizationId = await dispatcher.intent('SELECT_ORG', {
        organizations: organizationChoices
      })
    }
  }

  const space = await client.createSpace(
    {
      name,
      defaultLocale
    },
    organizationId
  )

  if (use) {
    await spaceUse({ spaceId: space.sys.id })
  }

  dispatcher.dispatch('SPACE_CREATED', { space })

  return space
}

module.exports.spaceCreate = spaceCreate

module.exports.handler = handle(spaceCreate)
