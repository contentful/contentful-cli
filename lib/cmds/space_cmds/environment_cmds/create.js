const Promise = require('bluebird')
const logging = require('../../../utils/log')
const { handleAsyncError: handle } = require('../../../utils/async')
const { createManagementClient } = require('../../../utils/contentful-clients')

module.exports.command = 'create'

module.exports.desc = 'Create an environment'

module.exports.builder = yargs => {
  return yargs
    .usage(
      "Usage: contentful space environment create --name 'Your Environment Name'"
    )
    .option('environment-id', {
      alias: 'e',
      describe: 'Id of the environment to create',
      demandOption: true
    })
    .option('name', {
      alias: 'n',
      describe: 'Name of the environment to create',
      demandOption: true
    })
    .option('source', {
      alias: 'src',
      describe:
        'ID of the source environment to create the new environment from',
      type: 'string'
    })
    .option('space-id', {
      alias: 's',
      describe: 'ID of the space that the environment will belong to',
      type: 'string'
    })
    .option('await-processing', {
      alias: 'w',
      describe: 'Wait until the environment is processed and ready',
      type: 'boolean',
      default: false
    })
    .option('management-token', {
      alias: 'mt',
      describe: 'Contentful management API token',
      type: 'string'
    })
    .epilog(
      [
        'See more at:',
        'https://github.com/contentful/contentful-cli/tree/master/docs/space/environment/create',
        'Copyright 2019 Contentful'
      ].join('\n')
    )
}

async function environmentCreate({
  context,
  name,
  source,
  awaitProcessing,
  environmentId
}) {
  const { managementToken, activeSpaceId } = context

  const client = await createManagementClient({
    accessToken: managementToken,
    feature: 'space-environment-create'
  })

  const space = await client.getSpace(activeSpaceId)

  const options = {}
  if (name) {
    options.name = name
  }

  const environment = source
    ? await space.createEnvironmentWithId(environmentId, options, source)
    : await space.createEnvironmentWithId(environmentId, options)

  logging.success(
    `Successfully created environment ${environment.name} (${
      environment.sys.id
    }) ${source ? `with source ${source}` : ''}`
  )

  if (awaitProcessing) {
    const DELAY = 3000
    const MAX_NUMBER_OF_TRIES = 10
    let count = 0

    logging.log('Waiting for processing...')

    while (count < MAX_NUMBER_OF_TRIES) {
      const status = (await space.getEnvironment(environment.sys.id)).sys.status
        .sys.id

      if (status === 'ready' || status === 'failed') {
        if (status === 'ready') {
          logging.success(
            `Successfully processed new environment ${environment.name} (${environment.sys.id})`
          )
        } else {
          logging.error('Environment creation failed')
        }
        break
      }

      await Promise.delay(DELAY)
      count++
    }
  }

  return environment
}

module.exports.environmentCreate = environmentCreate

module.exports.handler = handle(environmentCreate)
