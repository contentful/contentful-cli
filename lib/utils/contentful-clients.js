const { createClient } = require('contentful-management')
const { version } = require('../../package.json')
const { getContext } = require('../context')
const { agentFromProxy } = require('./proxy')

async function createManagementClient(params) {
  params.application = `contentful.cli/${version}`

  const context = await getContext()
  const { rawProxy, proxy, host } = context

  const proxyConfig = {}
  if (!rawProxy) {
    const { httpsAgent } = agentFromProxy(proxy)
    proxyConfig.httpsAgent = httpsAgent
  } else {
    proxyConfig.proxy = proxy
  }

  return createClient({ ...params, ...proxyConfig, host })
}

module.exports.createManagementClient = createManagementClient
