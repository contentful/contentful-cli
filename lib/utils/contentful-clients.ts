import {
  createClient,
  PlainClientAPI,
  ClientAPI,
  ClientOptions
} from 'contentful-management'
import { version } from '../../package.json'
import { getContext } from '../context'
import { agentFromProxy } from './proxy'

// Extending ClientOptions type but making specific adjustments for compatibility
type ClientParams = Omit<ClientOptions, 'apiAdapter' | 'accessToken'> & {
  accessToken?: string | (() => Promise<string>)
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  apiAdapter?: any // Using any here to match how the original JS file behaved
}

export async function createManagementClient(
  params: ClientParams
): Promise<ClientAPI> {
  params.application = `contentful.cli/${version}`

  const context = await getContext()
  const { rawProxy, proxy, host, insecure } = context

  const proxyConfig: Record<string, unknown> = {}
  if (!rawProxy) {
    const { httpsAgent } = agentFromProxy(proxy)
    proxyConfig.httpsAgent = httpsAgent
  } else {
    proxyConfig.proxy = proxy
  }

  return createClient({
    ...params,
    ...proxyConfig,
    host,
    insecure
  } as ClientOptions)
}

export async function createPlainClient(
  params: ClientParams,
  defaults: Record<string, unknown> = {}
): Promise<PlainClientAPI> {
  params.application = `contentful.cli/${version}`

  const context = await getContext()
  const { rawProxy, proxy, host, insecure } = context

  const proxyConfig: Record<string, unknown> = {}
  if (!rawProxy) {
    const { httpsAgent } = agentFromProxy(proxy)
    proxyConfig.httpsAgent = httpsAgent
  } else {
    proxyConfig.proxy = proxy
  }

  return createClient(
    {
      ...params,
      ...proxyConfig,
      host,
      insecure
    } as ClientOptions,
    { type: 'plain', defaults }
  ) as PlainClientAPI
}
