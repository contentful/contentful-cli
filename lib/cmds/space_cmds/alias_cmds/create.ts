import { Argv } from 'yargs'
import { handleAsyncError as handle } from '../../../utils/async'
import { createPlainClient } from '../../../utils/contentful-clients'
import { getHeadersFromOption } from '../../../utils/headers'
import logging from '../../../utils/log'

export const command = 'create'

export const desc = 'Create an environment alias'

export const builder = (yargs: Argv) => {
  return yargs
    .usage(
      'Usage: contentful space alias create --alias-id staging --target-environment-id dev-test'
    )
    .option('alias-id', {
      alias: 'a',
      describe: 'ID of the alias to create',
      demandOption: true
    })
    .option('target-environment-id', {
      alias: 'e',
      describe: 'ID of the target environment',
      type: 'string',
      demandOption: true
    })
    .option('space-id', {
      alias: 's',
      describe: 'ID of the space that the alias belongs to',
      type: 'string'
    })
    .option('management-token', {
      alias: 'mt',
      describe: 'Contentful management API token',
      type: 'string'
    })
    .option('header', {
      alias: 'H',
      type: 'string',
      describe: 'Pass an additional HTTP Header'
    })
}

interface Params {
  context: {
    managementToken: string
    activeSpaceId: string
  }
  aliasId: string
  targetEnvironmentId: string
  header?: string
}

export const environmentAliasCreate = async function ({
  context,
  aliasId,
  targetEnvironmentId,
  header
}: Params) {
  const { managementToken, activeSpaceId } = context

  const client = await createPlainClient({
    accessToken: managementToken,
    feature: 'space-environment-alias-create',
    headers: getHeadersFromOption(header)
  })

  const alias = await client.environmentAlias.createWithId(
    { spaceId: activeSpaceId, environmentAliasId: aliasId },
    {
      environment: {
        sys: {
          type: 'Link',
          linkType: 'Environment',
          id: targetEnvironmentId
        }
      }
    }
  )

  logging.success(
    `Successfully created environment alias ${alias.sys.id} targeting environment ${alias.environment.sys.id}`
  )

  return alias
}

export const handler = handle(environmentAliasCreate)
