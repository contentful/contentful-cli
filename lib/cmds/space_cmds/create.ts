import logging from '../../utils/log'
import { handleAsyncError as handle } from '../../utils/async'
import { createManagementClient } from '../../utils/contentful-clients'
import { spaceUse } from './use'

import { EventSystem } from '../../core/events'
import { CREATE_SPACE_HANDLER } from '../../core/events/scopes'

import { warningStyle } from '../../utils/styles'
import { confirmation } from '../../utils/actions'
import IntentSystem from '../../core/event-handlers/intents'
import LoggingSystem from '../../core/event-handlers/logging'

import createSpaceIntents from '../../core/event-handlers/intents/create-space-handler'
import createSpaceLogging from '../../core/event-handlers/logging/create-space-handler'

import { AbortedError } from '../../utils/aborted-error'
import { getHeadersFromOption } from '../../utils/headers'
import { copyright } from '../../utils/copyright'
import { Argv } from 'yargs'

export const command = 'create'

export const desc = 'Create a space'

export const builder = (yargs: Argv) => {
  return yargs
    .usage("Usage: contentful space create --name 'Your Space Name'")
    .option('name', {
      alias: 'n',
      describe: 'Name of the space to create',
      demandOption: true
    })
    .option('management-token', {
      alias: 'mt',
      describe: 'Contentful management API token',
      type: 'string'
    })
    .option('organization-id', {
      alias: 'org',
      describe: 'Organization owning the new space'
    })
    .option('yes', {
      alias: 'y',
      describe:
        'Confirm space creation without prompt, be aware this may result in extra monthly charges depend on your subscription'
    })
    .option('default-locale', {
      alias: 'l',
      describe: 'The default locale of the new space',
      type: 'string'
    })
    .option('use', {
      alias: 'u',
      describe:
        'Use the created space as default space when the --space-id is skipped.',
      type: 'boolean'
    })
    .option('header', {
      alias: 'H',
      type: 'string',
      describe: 'Pass an additional HTTP Header'
    })
    .epilog(copyright)
}

interface Context {
  managementToken?: string
}

interface SpaceCreateProps {
  context: Context
  name: string
  defaultLocale?: string
  yes?: boolean
  use?: boolean
  feature?: string
  organizationId?: string
  header?: string
}

export const spaceCreate = async function (argv: SpaceCreateProps) {
  const {
    context,
    name,
    defaultLocale,
    yes,
    use,
    header,
    feature = 'space-create'
  } = argv

  const { managementToken } = context
  let { organizationId = '' } = argv
  const client = await createManagementClient({
    feature,
    accessToken: managementToken,
    headers: getHeadersFromOption(argv.header)
  })

  logging.log(
    warningStyle(`Please be aware that adding new spaces to your subscription,
beyond the free ‘Micro’ space included in each subscription,
will result in extra monthly charges.
More information on available space types and their prices can be found on
the Pricing page: https://www.contentful.com/pricing/?faq_category=payments&faq=what-type-of-spaces-can-i-have#what-type-of-spaces-can-i-have`)
  )

  let confirm = false
  if (!yes) {
    confirm = await confirmation(`Do you want to confirm the space creation?`)
  } else {
    confirm = true
  }

  logging.log()

  if (!confirm) {
    logging.log(warningStyle(`Space creation aborted.`))
    throw new AbortedError()
  }

  const intentSystem = new IntentSystem()
  intentSystem.addHandler(createSpaceIntents())

  const loggingSystem = new LoggingSystem(logging)
  loggingSystem.addHandler(createSpaceLogging)

  const eventSystem = new EventSystem()
  eventSystem.attachSubsystem(intentSystem)
  eventSystem.attachSubsystem(loggingSystem)

  const dispatcher = eventSystem.dispatcher(CREATE_SPACE_HANDLER)

  if (!organizationId) {
    const result = await client.getOrganizations()

    if (result.items.length > 1) {
      dispatcher.dispatch('MULTIPLE_ORG_MEMBERSHIP')

      const organizationChoices = result.items
        .sort((a, b) => a.name.localeCompare(b.name))
        .map(
          organization => ({
            name: `${organization.name} (${organization.sys.id})`,
            value: organization.sys.id
          }),
          {}
        )

      organizationId = await dispatcher.intent('SELECT_ORG', {
        organizations: organizationChoices
      })
    }
  }

  const space = await client.createSpace(
    {
      name,
      defaultLocale
    },
    organizationId
  )

  if (use) {
    await spaceUse({ context, spaceId: space.sys.id, header })
  }

  dispatcher.dispatch('SPACE_CREATED', { space })

  return space
}

export const handler = handle(spaceCreate)
