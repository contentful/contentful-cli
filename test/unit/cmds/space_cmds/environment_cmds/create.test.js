import { environmentCreate } from '../../../../../lib/cmds/space_cmds/environment_cmds/create'
import {
  emptyContext,
  setContext
} from '../../../../../lib/context'
import { createManagementClient } from '../../../../../lib/utils/contentful-clients'

jest.mock('../../../../../lib/utils/contentful-clients')

const environmentData = {
  name: 'environment name',
  sys: {
    id: 'environmentId'
  }
}

const createEnvironmentWithIdStub = jest.fn().mockResolvedValue(environmentData)

const fakeClient = {
  getSpace: async () => ({
    createEnvironmentWithId: createEnvironmentWithIdStub
  })
}
createManagementClient.mockResolvedValue(fakeClient)

afterEach(() => {
  createEnvironmentWithIdStub.mockClear()
  createManagementClient.mockClear()
})
test('create environment - requires space id', async () => {
  emptyContext()
  setContext({
    cmaToken: 'mockedToken'
  })
  await expect(environmentCreate({})).rejects.toThrowErrorMatchingSnapshot()
  expect(createManagementClient).not.toHaveBeenCalled()
  expect(createEnvironmentWithIdStub).not.toHaveBeenCalled()
})

test('create new environment with id', async () => {
  emptyContext()
  setContext({
    cmaToken: 'mockedToken'
  })
  const result = await environmentCreate({
    spaceId: 'someSpaceID',
    environmentId: 'test'
  })
  expect(result).toBeTruthy()
  expect(createManagementClient).toHaveBeenCalledTimes(1)
  expect(createEnvironmentWithIdStub).toHaveBeenCalledTimes(1)
  expect(createEnvironmentWithIdStub.mock.calls[0][0]).toBe('test')
  expect(createEnvironmentWithIdStub.mock.calls[0][1]).toEqual({})
})

test('create new environment with id and name', async () => {
  emptyContext()
  setContext({
    cmaToken: 'mockedToken'
  })
  const result = await environmentCreate({
    spaceId: 'someSpaceID',
    environmentId: 'test',
    name: 'test'
  })
  expect(result).toBeTruthy()
  expect(createManagementClient).toHaveBeenCalledTimes(1)
  expect(createEnvironmentWithIdStub).toHaveBeenCalledTimes(1)
  expect(createEnvironmentWithIdStub.mock.calls[0][0]).toBe('test')
  expect(createEnvironmentWithIdStub.mock.calls[0][1]).toEqual({ name: 'test' })
})
