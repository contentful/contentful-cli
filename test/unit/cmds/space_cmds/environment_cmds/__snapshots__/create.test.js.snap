// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create environment - requires space id 1`] = `"You need to provide a space id. You can pass it via the [36m--space-id[39m parameter or by running [36mcontentful space use[39m"`;
